osbrain/agent.py

```diff
@@ -714,15 +720,24 @@ def _process_nonsub_event(self, socket_kind, socket, serialized):
         serialized : bytes
             Data received on the socket.
         """
-        message = pickle.loads(serialized)
+        serializer = self._get_serialization_from_socket(socket)
+
+        if serializer == 'pickle':
+            message = pickle.loads(serialized)
+        elif serializer == 'raw':
+            message = serialized
+
```

Comment:
This could be very easily moved to a new method, reducing complexity. In example:
 
    def _deserialize_message(socket, data):
        serializer = self._get_serialization_from_socket(socket)
        if serializer == 'raw':
            return data
        if serializer == 'pickle':
            return pickle.loads(data)
        raise ValueError('Serializer "%s" not supported!' % serializer)